### ISSUE TITLE
Attempting basic client sync example failed

### ISSUE THREAD

**Issue by saturn77** [slowtec/tokio-modbus#114]
In attempting to use the client-sync example, I simply cannot get it to work and was wondering if there is some library that need to include . . . Thanks, James

![image](https://user-images.githubusercontent.com/5698048/165869167-2a616b63-c851-40ff-af1c-f716af9816ae.png)

**Comment by uklotzde** [comment]
Please get familiar with the Rust module system: https://doc.rust-lang.org/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html

**Comment by saturn77** [comment]
The example will not compile, even with the Cargo.toml file set to include tokio-modbus. I have been using Rust for many years. This module is broken, and your repository does not have any continuous integration testing.

**Comment by uklotzde** [comment]
Ups, sorry. My fault, didn't look closely at the code and your previous experience :see_no_evil: 

The code you posted is in fact copied from one of the examples. All examples compile and work, verified by 2 CI workflows, namely [pre-commit](https://github.com/slowtec/tokio-modbus/blob/master/.github/workflows/pre-commit.yaml) and [continuous_integration](https://github.com/slowtec/tokio-modbus/blob/master/.github/workflows/continuous_integration.yaml). I checked both v0.5.2 and the main branch.

Did you enable both the `sync` and `tcp` features?

**Comment by uklotzde** [comment]
I just notices that the continuous_integration workflow is missing the `--all-targets` target selection. The examples are already checked in the pre-commit workflow by clippy, so this is not expected to reveal any hidden errors.

**Comment by saturn77** [comment]
Thanks for the feedback uklotzde. 

I was able to get the tokio-modbus library to work for me in a design and there were a couple of issues. 

1) Morphed the example code for tcp-sync with the  code in the document for tcp-sync and employed ? function
2) Added "features" of  sync and tcp in the Cargo.toml file ( Didn't see this in the Cargo.toml previoulsy)
3) Adjusted to using connect_slave as shown below. 

After this, I have been able to connect successfully. 

Thanks. James

![image](https://user-images.githubusercontent.com/5698048/166152101-15ab53b3-a6c1-4ec4-b415-e81ff1d337f0.png)
